src/dma.rs

```diff
@@ -0,0 +1,32 @@
+//! DMA Interface
+
+/// Static alias for DMA
+pub type Static<T> = &'static mut T;
+
+/// DMA Errors
+#[derive(Debug)]
+pub enum Error {
+    /// Previous data got overwritten before it could be read because it was
+    /// not accessed in a timely fashion
+    Overrun,
+    /// Transfer error
+    Transfer,
+}
+
+
+/// DMA Transfer future
+pub trait Transfer {
+    /// Return type
+    type Item;
+    /// Return type
+    type Payload;
+
+    /// Get buffer
+    fn deref(&self) -> &Self::Item;
```

Comment:
Does it make sense to allow access to the data while the transfer is still ongoing?